File: C/Assignment22/Program1.c
/////////////////////////////////////////////////////////////////////////////////////////////////////////
//  
// File name :   Program1.c
// Description : Accept character from user & check whether it is an alphabet or not.
// Author :      Pallavi Omprakash Malewar
// Date :        21/07/2026
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////


/*

Input :     F

Output :    TRUE

Input :     &

Output :    FALSE

*/


#include<stdio.h>
#include<stdlib.h>

#define TRUE 1
#define FALSE 0

typedef int BOOL;

BOOL CheckAlpha(char ch)
{
    if((ch <= 'Z' && ch >= 'A') || (ch >= 'a' && ch <= 'z'))
    {
        return TRUE;
    }
    return FALSE;
}

int main()
{
    char cValue = '\0';
    BOOL bRet = FALSE;

    printf("Enter the character : ");
    scanf("%c", &cValue);

    bRet = CheckAlpha(cValue);

    if(bRet == TRUE)
    {
        printf("It is Character\n");
    }
    else
    {
        printf("It is not a Character\n");
    }
    return 0;
}

File: JAVA/Assignment 10/Program1.java
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// Problem Statement : Write Java program which accept two strings from user & check   //
//                     whether first string is rotation of second string or not.       //
// Author : Pallavi Omprakash Malewar                                                  //
// Date : 4 Aug 2026                                                                   //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////


/*  

    Input : abcdefg  cdefgab

    Output: TRUE

*/


import java.util.Scanner;

class MyString
{
    public boolean CheckRotation(String s1, String s2)
    {
        int n1 = s1.length();
        int n2 = s2.length();

        if(n1 != n2) return false;

        String temp = s1 + s1;

        return temp.contains(s2);
    }
}

class Program1
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter First String : ");
        String s1 = sobj.nextLine();

        System.out.print("Enter Second String : ");
        String s2 = sobj.nextLine();

        MyString mobj = new MyString();

        boolean bRet = mobj.CheckRotation(s1, s2);

        if(bRet == true) System.out.println("TRUE");
        else System.out.println("FALSE");
    }
}
